TLM/TLM/UI/Textures/RoadSignThemeManager.cs

```diff
@@ -0,0 +1,241 @@
+namespace TrafficManager.UI.Textures {
+    using static TextureResources;
+    using System.Collections.Generic;
+    using System;
+    using System.Linq;
+    using CSUtil.Commons;
+    using TrafficManager.API.Traffic.Data;
+    using TrafficManager.API.Traffic.Enums;
+    using TrafficManager.Manager;
+    using TrafficManager.State;
+    using TrafficManager.State.ConfigData;
+    using TrafficManager.UI.SubTools;
+    using TrafficManager.UI.SubTools.SpeedLimits;
+    using TrafficManager.Util;
+    using UnityEngine;
+
+    /// <summary>
+    /// Singleton which manages road signs themes dynamically loaded and freed as the user .
+    /// The textures are loaded when OnLevelLoaded event is fired from the <see cref="Lifecycle.TMPELifecycle"/>.
+    /// </summary>
+    public class RoadSignThemeManager : AbstractCustomManager {
```

Comment:
I would run some code cleanup e.g: _Rider -> Code -> Rearrange code_ to organize things a bit because now everything is mixed. Some private fields don't follow naming convention (wrong case), some other could be `readonly` etc.
Sometimes I feel like line width is too small (in general, not here), especially in places with nested if's or debug logs where formatter break lines like mad creating walls of text for half page so 90% of time you scroll to search for the actual code 😂